### DISCUSSION TITLE
Add broker fees to flow

### DISCUSSION THREAD

**Discussion by razgraf** [sablier-labs/flow#242] [Ideas]
Fees for flow are now considered for the exit-points, specifically `withdraw`, instead of the old fee-model on `create`. Given the `withdraw` method is public, broker fees have been excluded from the mix, as passing them as a parameter could have been front-run (anyone can set themselves as the broker).

I'd like to propose adding the broker back, but setting it as an immutable parameter at stream creation. 

This way, when the stream is instantiated, the creator can pass an address and percentage as a "broker". This will be used at `withdraw` time, in perpetuity. 

Given flow streams are, by nature, meant to be long term, this can act as a big incentive for integrators to attract stream creators.

Related: #214, https://github.com/sablier-labs/v2-core/issues/1033

> [!NOTE]
>
> As we're quite ready for audits with flow, we can discuss this for an upcoming version of the protocol, post-launch.

**Comment by PaulRBerg** [comment]
Interesting idea but as discussed on the call, I'm against this proposal on the basis that:

1. I want us to get Flow out there and see how much interest there is from integrators.
2. We can always add this feature should it become clear that it's a dealbreaker for integrators.
3. We already have an incentive to sell integrations via protocol fees.
4. The implementation is not simple, and we want to wrap up development ASAP and freeze the code base for the audit.

**Reply by razgraf** [reply]
Points 1,2 and 4 have been addressed by my final note: discussing it at a later date. And of course, as you noted, based on demand/interest. 

But I don't understand point 3: how are integrators incentivized by **us** earning protocol fees?

**Reply by PaulRBerg** [reply]
Oh sorry, I missed the note in the OP!

> how are integrators incentivized by us earning protocol fees?

What I meant to say is that thanks to protocol fees, we now have a strong motivation to get Sablier integrated in other products. Before protocol fees, we didn't. @maxdesalle and I discussed this in other discussions (I can't remember which ones exactly).

**Reply by maxdesalle** [reply]
I think we mostly talked about it privately on Slack if I recall correctly @PaulRBerg, but I expanded on that argument in this discussion @razgraf (see the "Introducing a Fee Switch" section): https://github.com/sablier-labs/company-discussions/discussions/67

**Comment by PaulRBerg** [comment]
Closing as we ended up implementing a [`depositViaBroker`](https://github.com/sablier-labs/flow/blob/356d652382ca37acc46397155b3b98a62e61ad40/src/SablierFlow.sol#L295) function in Flow, which can be composed via `batch` to enable create+deposit+charge fee.
